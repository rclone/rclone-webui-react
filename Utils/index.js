import checkPropTypes from 'check-prop-types';
import {applyMiddleware, createStore} from 'redux';
import rootReducer from './../src/reducers';
import {middleware} from '../src/store';

export const findByTestAttr = (component, attr) => {
    const wrapper = component.find(`[data-test='${attr}']`);
    return wrapper;
};

export const checkProps = (component, expectedProps) => {
    const propsErr = checkPropTypes(component.propTypes, expectedProps, 'prop', component.name);
    return propsErr;
};

export const testStore = (initialState) => {
    const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);
    return createStoreWithMiddleware(rootReducer, initialState);
};