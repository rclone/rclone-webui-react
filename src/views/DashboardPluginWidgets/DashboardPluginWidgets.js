import React, {useEffect} from "react";
import * as PropTypes from "prop-types";
import {filterPluginsByType, getPluginBaseUrl, getPluginsArray} from "../../utils/PluginTools";
import pluginTypes from "../../pluginTypes";
import {Col, Row} from "reactstrap";
import DashboardPlugin from "../Base/DashboardPlugin/DashboardPlugin";
import {getIPAddress} from "../../utils/API/API";
import {connect} from "react-redux";
import {getPlugins} from "../../actions/pluginActions";

/**
 *
 * @param loadedPlugins {object}
 * @param getPlugins    {function}
 * @return {*}          {jsx}
 * @constructor
 */
function DashboardPluginWidgets({loadedPlugins, getPlugins}) {

    useEffect(() => {
        getPlugins();
    })

    let dashboardPlugins = getPluginsArray(loadedPlugins);
    dashboardPlugins = filterPluginsByType(dashboardPlugins, pluginTypes.DASHBOARD);
    return (
        <Row data-test={"dashboardPluginWidget"}>
            {dashboardPlugins.map(element => <Col lg={6} key={element.author + "/" + element.name}>
                <DashboardPlugin pluginUrl={getPluginBaseUrl(getIPAddress(), element.name, element.author)}/>
            </Col>)}
        </Row>
    )
}

const mapStateToProps = state => ({
    loadedPlugins: state.plugins.loadedPlugins,
});

DashboardPluginWidgets.propTypes = {
    loadedPlugins: PropTypes.object.isRequired,
    getPlugins: PropTypes.func.isRequired,
};

export default connect(mapStateToProps, {getPlugins})(DashboardPluginWidgets);