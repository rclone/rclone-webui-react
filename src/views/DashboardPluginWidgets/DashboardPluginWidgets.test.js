import React from "react";
import {shallow} from "enzyme";
import {findByTestAttr, testStore} from "../../../Utils";
import toJson from "enzyme-to-json";
import DashboardPluginWidgets from "./DashboardPluginWidgets";


const setUp = (intialState = {}, props = {}) => {
    const store = testStore(intialState);

    const component = shallow(<DashboardPluginWidgets {...props} store={store}/>);
    return component.childAt(0).dive();
};


describe('Dashboard Plugin Widget', function () {

    describe('renders', function () {
        let wrapper;
        beforeEach(() => {
            const initialState = {};

            const props = {};
            wrapper = setUp(initialState, props)
        });

        it('should render without crashing', function () {
            const component = findByTestAttr(wrapper, "dashboardPluginWidget");
            expect(component).toHaveLength(1);
        });

        it('should match snapshot', function () {
            expect(toJson(wrapper)).toMatchSnapshot()
        });
    });

});
