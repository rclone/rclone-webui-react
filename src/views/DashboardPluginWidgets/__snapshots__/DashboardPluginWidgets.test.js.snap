// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Plugin Widget renders should match snapshot 1`] = `
<Row
  data-test="dashboardPluginWidget"
  tag="div"
  widths={
    Array [
      "xs",
      "sm",
      "md",
      "lg",
      "xl",
    ]
  }
/>
`;
